lib/smart_answer/date_range.rb

```diff
@@ -72,7 +71,7 @@ def infinite?
     end
 
     def empty?
```

Comment:
I'm not sure I agree with this logic. Days fall between the two dates so it is wrong to say the range is empty.

Reply:
@andrewgarner sorry I don't understand, which days fall between the 2 dates?

Reply:
I was thinking in regards to ends on being before begins on. However, if were to use [Ruby's Range](https://ruby-doc.org/core-2.2.0/Range.html) library then it would be correct to say to is empty.

```
>> require 'date'
true
>> today = Date.today
#<Date: 2017-11-27 ((2458085j,0s,0n),+0s,2299161j)>
>> october = Date.new(2017, 10, 1)
#<Date: 2017-10-01 ((2458028j,0s,0n),+0s,2299161j)>
>> (october..today).to_a
[
    [ 0] #<Date: 2017-10-01 ((2458028j,0s,0n),+0s,2299161j)>,
    [ 1] #<Date: 2017-10-02 ((2458029j,0s,0n),+0s,2299161j)>,
    [ 2] #<Date: 2017-10-03 ((2458030j,0s,0n),+0s,2299161j)>,
    [ 3] #<Date: 2017-10-04 ((2458031j,0s,0n),+0s,2299161j)>,
    # ...
    [57] #<Date: 2017-11-27 ((2458085j,0s,0n),+0s,2299161j)>
]
>> (today..october).to_a
[]
```

Perhaps we should delegate `empty?` to a range?

Reply:
@andrewgarner but, if I understood well, when you define the range as `(october..today)` you are defining a different range then `(today..october)`. The first is not empty while the lattest is empty. So it should be coherent also with our implementation, isn't it?